### DISCUSSION TITLE
Should relationship IDs be stored in Mongo as BSON ObjectID instead of string?

### DISCUSSION THREAD

**Discussion by Moodycomputer** [payloadcms/payload#3019] [Feature Requests & Ideas]
When saving a relationship field, should the ID of the target record be stored as a BSON ObjectID instead of a string? This seems to be the recommended best practice from MongoDB, and is the practice adopted by Mongoose for [document population](https://mongoosejs.com/docs/populate.html).

Today, this field configuration—
```
{
  slug: 'example-collection',
  fields: [
    {
      name: 'owner',
      type: 'relationship',
      relationTo: 'users',
      hasMany: false,
    }
  ]
}
```

Saves to the database as—
```
{
  "owner": "6031ac9e1289176380734024"
}
```

Under this suggestion, that same configuration would save as 
```
{
  "owner": ObjectID("6031ac9e1289176380734024")
}
```

This would allow for sibling systems to access the Payload-connected database in a more standardized way, and allow for operations utilizing ObjectID helper functions within the database.

**Comment by jacobsfletch** [comment]
Hey @Moodycomputer! We've actually built a plugin to do exactly this, check it out: https://github.com/payloadcms/plugin-relationship-object-ids. The instructions in the README should be pretty explanatory but let me know if you have any questions.

**Reply by georgiev-anton** [reply]
Hello. Plugin not work with PayloadCMS v.2 (

https://github.com/payloadcms/plugin-relationship-object-ids/issues/2

**Comment by Moodycomputer** [comment]
@jacobsfletch This is exactly what I'm looking for, thank you!

Out of curiosity, why is this method not the default?

**Reply by jmikrut** [reply]
Because we have to be able to store custom text IDs and custom number IDs. Early on we made a decision to just store relationships as strings to make this all easier and more predictable but in retrospect, I think we could have probably just kept object IDs for sure. If this could become the default again in a future version.

**Reply by Moodycomputer** [reply]
That makes a lot of sense, the flexibility helps in some cases for sure. Thanks again!

**Comment by georgiev-anton** [comment]
@jacobsfletch Hi. help me please, how to fix  bug plugin-relationship-object-ids https://github.com/payloadcms/plugin-relationship-object-ids/issues/2 ?

**Comment by olivier5741** [comment]
Hi, it's hard to combine the following technologies on top of mongoDB because Payload does not support ObjectId relationships. The plugin is deprecated and I tried a pull request but there were issues : see https://github.com/payloadcms/plugin-relationship-object-ids/pull/5

* **Gatsbyjs** : supports **string** and **objectId**
* **MongoDB Atlas Charts** : support **objectId** only, need to turn string into objectId in the aggregation pipeline
* **PayloadCMS** : supports **string** only, plugin for supporting objectId is deprecated
* **Hasura** : supports **objectId** only, didn't find a work around so far ... (I just started integrating it)

I'm using PayloadCMS over 30 collections with many relationships and this is the only thing that's in my way. All the rest is super smooth ;).

**Reply by olivier5741** [reply]
This is the best to make every tool happy. A string id for payload under `${concept.name}` and an object id under `${concept.name} + "Id"` for the others ... Overkill but effective :)

<img width="469" alt="Screenshot 2024-05-16 at 13 09 36" src="https://github.com/payloadcms/payload/assets/3897291/3bb8351a-5dff-4bd7-ad67-46e778fc32a8">
